app/controllers/actions/CheckEmailVerificationFilter.scala

```diff
@@ -44,21 +46,22 @@ class CheckEmailVerificationFilterImpl(mode: Option[Mode],
     if (frontendAppConfig.emailVerificationEnabled && !mode.contains(AmendMode)) {
       request.userAnswers.get(BusinessContactDetailsPage) match {
         case Some(contactDetails) =>
-          emailVerificationService.isEmailVerified(contactDetails.emailAddress, request.userId).map {
+          emailVerificationService.isEmailVerified(contactDetails.emailAddress, request.userId).flatMap {
             case Verified =>
               logger.info("CheckEmailVerificationFilter - Verified")
-              None
+              Future(Option.empty[Result])
             case LockedTooManyLockedEmails =>
               logger.info("CheckEmailVerificationFilter - LockedTooManyLockedEmails")
-              Some(Redirect(controllers.routes.EmailVerificationCodesAndEmailsExceededController.onPageLoad().url))
-
+              Future(Some(Redirect(controllers.routes.EmailVerificationCodesAndEmailsExceededController.onPageLoad().url)))
```

Comment:
`Future.successful`

Reply:
We shouldn't force `Future.successful` as anything can happen. When we invoke a `Future` your saying to allocate this operation to a new Thread. This could always fail due to a variety of reasons such as the cpu not being able to allocate execution time and thread starvation. There is never a guarantee a thread will be executed as expected. So sending a successful signal might be inaccurate. We should only do this in tests. 

Is there specific benefit of having `Future.successful` I am not aware of ? Because this will be the result anyway when the operation is executed successfully by the execution context.

Reply:
Ignore Future.successful is already completed just a wrapper

Reply:
Exactly `Future.successful` does not allocate a new Thread. I see `Some(Redirect(controllers.routes.EmailVerificationCodesAndEmailsExceededController.onPageLoad().url))` as a literal value that will not throw any Exceptions and is not blocking